content/deyemiobaa/building-a-todolist-with-rails.md

```diff
@@ -0,0 +1,351 @@
+2. Convention over Configuration: This is one of Rails’ superpowers. Rails has a set of guidelines and format for doing things. From naming methods, classes, files, and building a database schema. These conventions help to keep the whole application in sync and help you spend less time debugging errors that result from the wrong configuration.
+
+## Setting up your dev environment: Installing Ruby, Rails, and PostgreSQL
+
+[GoRails](https://goRails.com/) Has in-depth guides on setting up Rails on various machines. Here's a link to all of them. Since we're using PostgreSQL for this project, skip any step that mentions setting up SQLite or MySQL.
+
+- [For Ubuntu](https://goRails.com/setup/ubuntu/22.04)
+- [Mac OS](https://goRails.com/setup/macos/13-ventura)
+- [For Windows](https://www.hanselman.com/blog/ruby-on-Rails-on-windows-is-not-just-possible-its-fabulous-using-wsl2-and-vs-code)
+
+## Setting up a new Rails app
+
+To create a new Rails app, simply open up your terminal, navigate to where you want this app to stay and run the following command.
+
+```bash
+rails new todolist -d postgresql –css tailwind
+```
+
+The commands after the app name (`todolist`) are optional. The `-d postgresql` flag tells rails to use PostgreSQL as the database for this app. The `–css tailwind` flag tells rails to use tailwind as the CSS framework for this app. You can use any CSS framework you want or just plain CSS. I chose tailwind because it’s easy to use and it’s a breeze to set up.
+
+After running the command, you’ll see a bunch of files and folders being created. This is the default structure of a Rails app. You can read more about it [here](https://guides.rubyonrails.org/getting_started.html#creating-the-blog-application).
+
+To get our app running, we need to make sure it is connected to a database. The database config lives in the `config/database.yml` file. There we can see the provided database name and some other info. This database doesn't exist yet and needs to be created. For most people, you need to supply your PostgreSQL username and password that was set during installation to this file. The default username is `postgres` and this applies to most Linux distros.
+
+Most times we would use an env file to manage this but we can keep it there for the scope of this project.
+
+```yml
+...
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  username: postgres
+  password: yourpassword
+...
+```
+
+To create the database, run `rails db:create`.
+
+Now run `rails s` to start the server. If everything went well, you should see a url to visit (`http://127.0.0.1:3000/`) in your terminal. Open that url in your browser and you should see the default Rails page.
+
+Since we have TailwindCSS in our project, starting our server requires an additional step. We need to run `rails tailwindcss:watch` in a separate terminal window. This will compile our tailwind styles and make them available to our app.
```

Comment:
`rails tailwindcss:watch` in a new terminal window does not do anything but spawn the rails help menu.

Reply:
That's a strange behaviour. May I ask what OS you're running this project on?

that command should produce something like this
![image](https://user-images.githubusercontent.com/55185309/224572813-d35244e5-740c-419d-bf0c-e8aaaf56ee6f.png)

Reply:
Can you try `bin/rails tailwindcss:watch`?

Reply:
Do I need to cd to the app directory to do this? If so, it should be mentioned in the tutorial.

Reply:
I tried this in the app directory and I get the following error:
```
rails aborted!
Don't know how to build task 'tailwindcss:watch' (See the list of available tasks with `rails --tasks`)

(See full trace by running task with --trace)
```

Reply:
The command should be run in the root of the project directory. Also, It seems some dependencies didn't install correctly. Can you run `bundle update` and then try again?

Reply:
I tried this and still get the same error.